polyply/tests/test_random_walk.py

```diff
@@ -14,84 +14,300 @@
+
+@pytest.mark.parametrize('build_attr, pos, start, npos', (
+    # simple test; create all coordinates
+    ({0: True, 1: True, 2: True, 3: True,
+      4: True, 5: True, 6: True, 7: True, 8: True, 9: True},
+     None,
+     None,
+     0),
+    # start in the middle of the chain
+    ({0: True, 1: True, 2: True, 3: True,
+      4: True, 5: True, 6: True, 7: True, 8: True, 9: True},
+     None,
+     5,
+     0),
+    # here we look for a starting point and build the rest
+    ({0: False, 1: False, 2: True, 3: True,
+      4: True, 5: True, 6: True, 7: True, 8: True, 9: True},
+     np.array([[1.0, 1.0, 0.67],
+               [1.0, 1.0, 1.37]]),
+     None,
+     2,
+     ),
+    # here we need to skip one already defined position
+    ({0: False, 1: True, 2: False, 3: True,
+      4: True, 5: True, 6: True, 7: True, 8: True, 9: True},
+     np.array([[1.0, 1.0, 0.67],
+               [np.inf, np.inf, np.inf],
+               [1.0, 1.0, 1.30]]),
+     None,
+     3),
+    # here we trigger a rewind
+   # ({0: False, 1: False, 2: False, 3: False,
+   #   4: False, 5: False, 6: False, 7: True, 8: True, 9: True},
+   #  np.array([[1.0, 1.0, 0.67],
+   #            [1.0, 1.0, 1.37],
+   #            [1.0, 1.37, 1.0],
+   #            [1.37, 1.0, 1.0],
+   #            [1.0, 0.63, 1.0],
+   #            [0.63, 1.0, 1.0],
+   #            [1.0, 1.0, 1.0],
+   #            ]),
+   #  None,
+   #  7),
+     ))
+def test_run_molecule(nonbond_matrix, molecule, build_attr, npos, pos, start):
+    # add positions of the rest of the chain
+    nb_matrix = add_positions(nonbond_matrix, npos, pos=pos)
+    # create instance of processor
+    vector_bundle = polyply.src.linalg_functions.norm_sphere(500)
+    proccessor = RandomWalk(mol_idx=0,
+                            nonbond_matrix=nb_matrix,
+                            maxdim=np.array([10., 10., 10.]),
+                            max_force=100.0,
+                            maxiter=49,
+                            vector_sphere=vector_bundle,
+                            start_node=start)
+    # set molecule attribute which is normally set by the run_molecule class
+    nx.set_node_attributes(molecule, build_attr, "build")
+    proccessor.run_molecule(molecule)
+    for pos in proccessor.nonbond_matrix.positions:
+        assert all(pos != np.array([np.inf, np.inf, np.inf]))
+
+
+@pytest.mark.parametrize('point, old_point, node_dict, expected', (
+    # basic example
+    (np.array([1., 1., 2.]),
+     np.array([1., 1., 1.]),
+     {"rw_options": [[np.array([0., 0., 1.0]), 90.0]]},
+     True),
+    # false because goes back
+    (np.array([1., 1., 0.5]),
+     np.array([1., 1., 1.]),
+     {"rw_options": [[np.array([0., 0., 1.0]), 90.0]]},
+     False),
+    # false because angle not large enough
+    (np.array([1.5, 1.5, 1.5]),
+     np.array([1., 1., 1.]),
+     {"rw_options": [[np.array([0., 0., 1.0]), 50.0]]},
+     False),
 ))
```

Comment:
How about adding e.g. parallel vectors, or zero length vectors?

Reply:
zero length vectors are not generated;

Reply:
What type of parallel vectors do you mean? Like parallel to what?

Reply:
So that the vector (a - b) parallel is to the normal vector